common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -116,18 +121,28 @@ private static void schedule(Thread thread, Runnable task, boolean isWatch) {
      *
      * @return {@code true} if and only if the watcher thread has been terminated
      */
-    public static boolean awaitInactivity(long timeout, TimeUnit unit) throws InterruptedException {
-        if (unit == null) {
-            throw new NullPointerException("unit");
-        }
-
-        Thread watcherThread = ThreadDeathWatcher.watcherThread;
-        if (watcherThread != null) {
-            watcherThread.join(unit.toMillis(timeout));
-            return !watcherThread.isAlive();
-        } else {
-            return true;
+    public static boolean awaitInactivity(long amount, TimeUnit unit) throws InterruptedException {
```

Comment:
@Scottmitch need to think about this a bit more... That said let me factor out the Queue replacement to another PR as I think we want to pull this in in all cases.